### ISSUE TITLE
`rumdl-pre-commit`: `pre-commit run rumdl --all-files` runs `rumdl` in multiple batches, output looks inconsistent vs `rumdl check .`

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#350]
## Summary

Running `rumdl` directly vs via the official pre-commit hook repo produces confusing, seemingly inconsistent summaries on the same codebase and the same `rumdl` version.

`uvx rumdl@latest check .` reports a single aggregated result, while `pre-commit run rumdl --all-files` (and `prek run rumdl --all-files`) appears to invoke `rumdl` multiple times in batches (8 files per run), mixing `Success` and `Issues` summaries. This makes it look like results differ between runs even when nothing changed.

## Steps to reproduce

1. In a repo with `rumdl` issues, run:
   ```bash
   uvx rumdl@latest check .
   ```

2. Without modifying anything, run:
   ```bash
   pre-commit run rumdl --all-files
   ```
   (Same behavior with `prek run rumdl --all-files`.)

## Actual output

Direct run:
```text
Issues: Found 97 issues in 7/71 files (46ms)
Run `rumdl fmt` to automatically fix 97 of the 97 issues
```

Pre-commit (same codebase, same `rumdl` version):
```text
Success: No issues found in 8 files (73ms)
Success: No issues found in 8 files (48ms)
Issues: Found 17 issues in 2/8 files (73ms)
Run `rumdl fmt` to automatically fix 17 of the 17 issues
Success: No issues found in 8 files (77ms)
Success: No issues found in 8 files (67ms)
Success: No issues found in 8 files (54ms)
Issues: Found 12 issues in 2/8 files (73ms)
Run `rumdl fmt` to automatically fix 12 of the 12 issues
Issues: Found 67 issues in 2/8 files (76ms)
Run `rumdl fmt` to automatically fix 67 of the 67 issues
Issues: Found 1 issues in 1/7 files (68ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

## Expected behavior

One of the following (either is fine):

- The pre-commit hook runs `rumdl` once for `--all-files` and reports a single aggregated summary similar to `rumdl check .`, or
- If batching is required, the hook prints a final aggregated summary so it is clear the overall results match the direct run.

**Comment by rvben** [comment]
This should be fixed in v0.1.5. Thanks for reporting @pygarap!

Can you verify this works correctly with the latest release?

**Comment by pygarap** [comment]
@rvben where did you see that `ruff-pre-commit` is using the `--quite` flag?

![image](https://github.com/user-attachments/assets/cf67a52e-8a76-4c88-9ae5-9dd7fc197844)

**Comment by rvben** [comment]
Good catch @pygarap, thanks for noticing. There clearly was a small hallucination there, as ruff doesn't use `--quiet`. Looking at [ruff-pre-commit](https://github.com/astral-sh/ruff-pre-commit/blob/main/.pre-commit-hooks.yaml) and [black-pre-commit](https://github.com/psf/black-pre-commit-mirror/blob/main/.pre-commit-hooks.yaml), they both use `require_serial: true` which tells pre-commit to pass all files in a single invocation.

I've updated rumdl-pre-commit (890c9f5) to:
- Add `require_serial: true` 
- Remove `--quiet` so you get the helpful summary

This should now give you a single aggregated output matching `rumdl check .`.

To get the fix, update your `.pre-commit-config.yaml` to use the latest rev (or any rev after this fix).

Thanks for the follow-up!
